tests/libvmifact/factory.go

```diff
@@ -146,3 +148,23 @@ func NewWindows(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 	vmi.Spec.Domain.Firmware = &kvirtv1.Firmware{UUID: WindowsFirmware}
 	return vmi
 }
+
+func NewTinyOSVMWithDataVolume(dv *cdiv1.DataVolume, extraVMOpts ...libvmi.VMOption) *kvirtv1.VirtualMachine {
```

Comment:
I'm sorry, I have hard time to convince myself it fits a factory.

- The name tells me little about what this is representing. It is tiny because it uses low memory? small image? low amount of added options?
- `WithDataVolume` in the name hints it is not confirming to the builder pattern, but to the "name joiner". Based on this patter, we can also say `..WithResourceMemoryWithOneInterface...".

Reply:
> * The name tells me little about what this is representing. It is tiny because it uses low memory? small image? low amount of added options?

I am trying to encapsulate NewPersistentCirros/NewPersistentAlpine under one umbrella so tiny was what I came up with. I mean minimal Linux distros like Alpine/Cirros, but I am open to other suggestions

Reply:
> * `WithDataVolume` in the name hints it is not confirming to the builder pattern, but to the "name joiner".

Yeah not the right choice. I do think it fits a factory it's just confusing naming - I could go with `NewPersistentDiskTinyOS` or something like that. Naming is not fun lol

Reply:
I do not understand why you need to have a factory for this.
If you want to add a persisted-something, just add it... You will violate nothing by repeating the building blocks. If there is no good name, it means it is not a fit for a factory, that is my rule.

Reply:
Yeah TBH if you recall this was my initial worry @dankenigsberg @iholder101 
But having said that the creation of such a VM is really bloated compared to others,
so that, together with us having a place for NewCirros/NewAlpine, I think we can consider it (doesn't have to be under vmfact)

Reply:
wdyt @EdDev

Reply:
> If you create a factory in a specific test, then I see no problem.

Unfortunately, the need arose in some PRs to have this exposed centrally
(#this, #11488, #11621)
Would you totally object such a factory?
The alternative would be to duplicate the (bloated) creation of such VMs all over which isn't very nice.
@dankenigsberg @iholder101 WDYT?

Reply:
I don't know. I agree that `NewTinyOSVMWithDataVolume`'s cumbersome name suggests that it is a specific entity. On the other hand, it is clearly used by many tests. I think that if it was my PR, I would define this factory in `libstorage` or in somewhere else under `storage`, as there is where most of the users sit.

Reply:
> I would define this factory in `libstorage` or in somewhere else under `storage`, as there is where most of the users sit.

Are you suggesting others will import from under `storage` or just copy over the definitions?